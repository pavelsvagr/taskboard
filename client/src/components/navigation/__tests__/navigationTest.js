/* eslint react/jsx-filename-extension:0 */
import React from "react"
import ReactDOM from "react-dom"
import fetchMock from "fetch-mock"
import thunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import Navigation from "../Navigation"
import testStore from "../../../helpers/tests/testAdminStore"
import testUserStore from "../../../helpers/tests/testUserStore"
import windowMedia from "../../../helpers/tests/windowMedia"

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe("Navigation.jsx", () => {
  afterEach(() => {
    fetchMock.restore()
  })

  it("SMOKE: render", () => {
    const store = mockStore(testStore)
    Object.defineProperty(window, "matchMedia", windowMedia)

    const div = document.createElement("div")
    ReactDOM.render(
      <Provider store={store}>
        <MemoryRouter>
          <Navigation location={{ pathname: "" }} />
        </MemoryRouter>
      </Provider>, div)
  })

  it("Snapshot: Match admin", () => {
    const store = mockStore(testStore)
    Object.defineProperty(window, "matchMedia", windowMedia)

    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Navigation location={{ pathname: "" }} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("Snapshot: Match user", () => {
    const store = mockStore(testUserStore)
    Object.defineProperty(window, "matchMedia", windowMedia)

    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Navigation location={{ pathname: "" }} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})