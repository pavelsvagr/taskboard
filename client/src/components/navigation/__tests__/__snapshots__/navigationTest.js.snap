// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.jsx Snapshot: Match admin 1`] = `
Array [
  <div
    className="nav__button"
  >
    <div
      className=""
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-label="menu"
          className="anticon anticon-menu"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  <aside
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed ant-layout-sider-has-trigger nav"
    style={
      Object {
        "flex": "0 0 80px",
        "height": "100vh",
        "left": 0,
        "maxWidth": "80px",
        "minWidth": "80px",
        "overflow": "auto",
        "position": "fixed",
        "width": "80px",
        "zIndex": 90,
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="nav__button"
        onClick={[Function]}
        style={
          Object {
            "position": "absolute",
            "right": 30,
            "top": 10,
          }
        }
      >
        <div
          className=""
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg ant-btn-background-ghost"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-label="close"
              className="anticon anticon-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="text-center"
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <a
          href="/boards"
          onClick={[Function]}
        >
          <div
            className="logo"
          />
        </a>
      </div>
      <ul
        className="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={
          Object {
            "paddingBottom": 40,
          }
        }
      >
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="appstore"
            className="anticon anticon-appstore"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="appstore"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zM464 544H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H212V612h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/boards"
              onClick={[Function]}
            >
              Boards
            </a>
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="team"
            className="anticon anticon-team"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="team"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M824.2 699.9a301.55 301.55 0 00-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 00-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 008 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 01612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 008-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 01-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 01612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 01-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 008 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/teams"
              onClick={[Function]}
            >
              Teams
            </a>
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="user"
            className="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/users"
              onClick={[Function]}
            >
              Users
            </a>
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="key"
            className="anticon anticon-key"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="key"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5l-41.1 41.1-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-44.9 44.9-62.3-62.3a8.15 8.15 0 00-11.4 0l-39.8 39.8a8.15 8.15 0 000 11.4l62.3 62.3-65.3 65.3a8.03 8.03 0 000 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6A304.06 304.06 0 00608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644c-60.9 0-118.2-23.7-161.2-66.8-43.1-43-66.8-100.3-66.8-161.2 0-60.9 23.7-118.2 66.8-161.2 43-43.1 100.3-66.8 161.2-66.8 60.9 0 118.2 23.7 161.2 66.8 43.1 43 66.8 100.3 66.8 161.2 0 60.9-23.7 118.2-66.8 161.2z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/credentials"
              onClick={[Function]}
            >
              Credentials
            </a>
          </span>
        </li>
        <div
          className="text-center"
          style={
            Object {
              "margin": 10,
            }
          }
        >
          <span
            className="ant-avatar ant-avatar-circle nav__avatar "
            style={Object {}}
          >
            <span
              className="ant-avatar-string"
              style={
                Object {
                  "WebkitTransform": "scale(1) translateX(-50%)",
                  "msTransform": "scale(1) translateX(-50%)",
                  "transform": "scale(1) translateX(-50%)",
                }
              }
            >
              T
            </span>
          </span>
        </div>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="bell"
            className="anticon anticon-bell"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="bell"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M816 768h-24V428c0-141.1-104.3-257.8-240-277.2V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.8C336.3 170.2 232 286.9 232 428v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48z"
              />
            </svg>
          </span>
          <span>
            Notifications
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="poweroff"
            className="anticon anticon-poweroff"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="poweroff"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M705.6 124.9a8 8 0 00-11.6 7.2v64.2c0 5.5 2.9 10.6 7.5 13.6a352.2 352.2 0 0162.2 49.8c32.7 32.8 58.4 70.9 76.3 113.3a355 355 0 0127.9 138.7c0 48.1-9.4 94.8-27.9 138.7a355.92 355.92 0 01-76.3 113.3 353.06 353.06 0 01-113.2 76.4c-43.8 18.6-90.5 28-138.5 28s-94.7-9.4-138.5-28a353.06 353.06 0 01-113.2-76.4A355.92 355.92 0 01184 650.4a355 355 0 01-27.9-138.7c0-48.1 9.4-94.8 27.9-138.7 17.9-42.4 43.6-80.5 76.3-113.3 19-19 39.8-35.6 62.2-49.8 4.7-2.9 7.5-8.1 7.5-13.6V132c0-6-6.3-9.8-11.6-7.2C178.5 195.2 82 339.3 80 506.3 77.2 745.1 272.5 943.5 511.2 944c239 .5 432.8-193.3 432.8-432.4 0-169.2-97-315.7-238.4-386.7zM480 560h64c4.4 0 8-3.6 8-8V88c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/api/logout"
            >
              Logout
            </a>
          </span>
        </li>
      </ul>
    </div>
    <div
      className="ant-layout-sider-trigger"
      onClick={[Function]}
      style={
        Object {
          "width": "80px",
        }
      }
    >
      <span
        aria-label="right"
        className="anticon anticon-right"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
    </div>
  </aside>,
]
`;

exports[`Navigation.jsx Snapshot: Match user 1`] = `
Array [
  <div
    className="nav__button"
  >
    <div
      className=""
    >
      <button
        className="ant-btn ant-btn-primary ant-btn-lg"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-label="menu"
          className="anticon anticon-menu"
          role="img"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="menu"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  <aside
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed ant-layout-sider-has-trigger nav"
    style={
      Object {
        "flex": "0 0 80px",
        "height": "100vh",
        "left": 0,
        "maxWidth": "80px",
        "minWidth": "80px",
        "overflow": "auto",
        "position": "fixed",
        "width": "80px",
        "zIndex": 90,
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="nav__button"
        onClick={[Function]}
        style={
          Object {
            "position": "absolute",
            "right": 30,
            "top": 10,
          }
        }
      >
        <div
          className=""
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg ant-btn-background-ghost"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-label="close"
              className="anticon anticon-close"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="text-center"
        style={
          Object {
            "margin": 10,
          }
        }
      >
        <a
          href="/boards"
          onClick={[Function]}
        >
          <div
            className="logo"
          />
        </a>
      </div>
      <ul
        className="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={
          Object {
            "paddingBottom": 40,
          }
        }
      >
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="appstore"
            className="anticon anticon-appstore"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="appstore"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zM464 544H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H212V612h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/boards"
              onClick={[Function]}
            >
              Boards
            </a>
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="team"
            className="anticon anticon-team"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="team"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M824.2 699.9a301.55 301.55 0 00-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 00-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 008 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 01612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 008-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 01-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 01612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 01-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 008 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
              />
            </svg>
          </span>
          <span>
            <a
              aria-current={null}
              href="/teams"
              onClick={[Function]}
            >
              Teams
            </a>
          </span>
        </li>
        <div
          className="text-center"
          style={
            Object {
              "margin": 10,
            }
          }
        >
          <span
            className="ant-avatar ant-avatar-circle nav__avatar "
            style={Object {}}
          >
            <span
              className="ant-avatar-string"
              style={
                Object {
                  "WebkitTransform": "scale(1) translateX(-50%)",
                  "msTransform": "scale(1) translateX(-50%)",
                  "transform": "scale(1) translateX(-50%)",
                }
              }
            >
              T
            </span>
          </span>
        </div>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="bell"
            className="anticon anticon-bell"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="bell"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M816 768h-24V428c0-141.1-104.3-257.8-240-277.2V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.8C336.3 170.2 232 286.9 232 428v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48z"
              />
            </svg>
          </span>
          <span>
            Notifications
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <span
            aria-label="poweroff"
            className="anticon anticon-poweroff"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="poweroff"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M705.6 124.9a8 8 0 00-11.6 7.2v64.2c0 5.5 2.9 10.6 7.5 13.6a352.2 352.2 0 0162.2 49.8c32.7 32.8 58.4 70.9 76.3 113.3a355 355 0 0127.9 138.7c0 48.1-9.4 94.8-27.9 138.7a355.92 355.92 0 01-76.3 113.3 353.06 353.06 0 01-113.2 76.4c-43.8 18.6-90.5 28-138.5 28s-94.7-9.4-138.5-28a353.06 353.06 0 01-113.2-76.4A355.92 355.92 0 01184 650.4a355 355 0 01-27.9-138.7c0-48.1 9.4-94.8 27.9-138.7 17.9-42.4 43.6-80.5 76.3-113.3 19-19 39.8-35.6 62.2-49.8 4.7-2.9 7.5-8.1 7.5-13.6V132c0-6-6.3-9.8-11.6-7.2C178.5 195.2 82 339.3 80 506.3 77.2 745.1 272.5 943.5 511.2 944c239 .5 432.8-193.3 432.8-432.4 0-169.2-97-315.7-238.4-386.7zM480 560h64c4.4 0 8-3.6 8-8V88c0-4.4-3.6-8-8-8h-64c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8z"
              />
            </svg>
          </span>
          <span>
            <a
              href="/api/logout"
            >
              Logout
            </a>
          </span>
        </li>
      </ul>
    </div>
    <div
      className="ant-layout-sider-trigger"
      onClick={[Function]}
      style={
        Object {
          "width": "80px",
        }
      }
    >
      <span
        aria-label="right"
        className="anticon anticon-right"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="right"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
          />
        </svg>
      </span>
    </div>
  </aside>,
]
`;
